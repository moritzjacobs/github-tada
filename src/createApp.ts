import createButton from "./createButton";
import { Comment } from "./comments";
import { createElement } from "./DOM";

type Props = {
	comments: Array<Comment>;
};

const createApp = ({ comments }: Props) => {
	if (comments.length < 1) {
		return null;
	}

	const commentEls = comments.map((c) => (c.target ? createButton(c) : null));

	return [
		createElement(
			"div",
			{
				className: "discussion-sidebar-heading text-bold",
			},
			"Browse by reaction"
		),
		...commentEls,
	];
};

export default createApp;
