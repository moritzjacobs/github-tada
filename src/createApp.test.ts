import createApp from "./createApp";

const reactions = [
	{
		emoji: "ðŸ’¯",
		type: "100",
		count: 23,
	},
];

const target = document.createElement("div");

target.dataset.gid = "1234";

describe("createApp()", () => {
	it("renders correctly when there are no items", () => {
		const tree = createApp({ comments: [] });

		expect(tree).toMatchSnapshot();
	});

	it("renders correctly when there are items", () => {
		const tree = createApp({ comments: [{ target, reactions }] });

		expect(tree).toMatchSnapshot();
	});
});
