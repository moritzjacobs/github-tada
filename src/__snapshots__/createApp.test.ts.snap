// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createApp() renders correctly when there are items 1`] = `
Array [
  <div
    class="discussion-sidebar-heading text-bold"
  >
    Browse by reaction
  </div>,
  <button
    class="grb-button"
    type="button"
  >
    <span
      class="grb-reaction"
    >
      ðŸ’¯Â 23
    </span>
  </button>,
]
`;

exports[`createApp() renders correctly when there are no items 1`] = `null`;
